app/src/main/java/com/example/concordia_campus_guide/LocationFragment/LocationFragmentViewModel.java

```diff
@@ -51,14 +54,126 @@ private GeoJsonLayer initLayer(GoogleMap map, Context applicationContext){
         GeoJsonLayer layer = null;
         try {
             layer = new GeoJsonLayer(map, R.raw.buildingcoordinates, applicationContext);
-        } catch (IOException e) {
-            e.printStackTrace();
-        } catch (JSONException e) {
+        } catch (IOException | JSONException e) {
             e.printStackTrace();
         }
         return layer;
     }
 
+    /**
+     * Generate the hall building overlays
+     * @return the generate ground overlay option
+     */
+    public GroundOverlayOptions getHallBuildingOverlay(){
+        return new GroundOverlayOptions()
+                .position(new LatLng(45.4972685, -73.5789475), (float) 68, (float) 68)
+                .image(BitmapDescriptorFactory.fromResource(R.drawable.hall_9))
+                .bearing((float) 34);
+    }
+
+    /**
+     * @param groundOverlay
+     * @param floorNb
+     */
+    public void setHallFloorplan(GroundOverlay groundOverlay, int floorNb){
+        if(floorNb == 8)  setHall8Floorplan(groundOverlay);
+        else if(floorNb == 9) setHall9Floorplan(groundOverlay);
+    }
+
+    /**
+     * set ground overlay image for Hall 8th building
+     * @param groundOverlay
+     */
+    private void setHall8Floorplan(GroundOverlay groundOverlay){
+        groundOverlay.setImage(BitmapDescriptorFactory.fromResource(R.drawable.hall_8));
+    }
+
+    /**
+     * set ground overlay image for Hall 9th building
+     * @param groundOverlay
+     */
+    private void setHall9Floorplan(GroundOverlay groundOverlay){
+        groundOverlay.setImage(BitmapDescriptorFactory.fromResource(R.drawable.hall_9));
+    }
+
+    /**
+     * Generate the John Molson  building overlays
+     * @return the generate ground overlay option
+     */
+    public GroundOverlayOptions getMBBuildingOverlay(){
+        //400 x 570 / 645 x 645
+        //actual dimensions 22.58 x 64 -> 36.41 x 72
+        return new GroundOverlayOptions()
+                //45.495245, -73.578941
+                .position(new LatLng( 45.495305, -73.578885), (float) 65, (float) 65)
+                .image(BitmapDescriptorFactory.fromResource(R.drawable.mb_1))
+                .bearing((float) 127);
+    }
+
+    /**
+     * set ground overlay image for Mb floors
+     * @param groundOverlay
+     * @param floorNb
+     */
+    public void setMBFloorplan(GroundOverlay groundOverlay, int floorNb){
+        if(floorNb == -1)  setMBS2Floorplan(groundOverlay);
+        else if(floorNb == 1)  setMB1Floorplan(groundOverlay);
+    }
+
+    /**
+     * set ground overlay image for Mb S2 floor
+     * @param groundOverlay
+     */
+    private void setMBS2Floorplan(GroundOverlay groundOverlay){
```

Comment:
I am concerned with this type of method. It seems every time we will want to add a floor, we will have to add a function in the code. This does not seem scalable as we add more floors. I think it would make sense to have a single parametrized method that sets a different image for the ground overlay based on the parameter. 

For example, for our info card, we added the building images using a single method, that took in a parameter of the building code and fetched the appropriate image. https://github.com/Concordia-Campus-Guide/Concordia-Campus-Guide/blob/6d2b4320216b184915428c51e7130f846ece6d7c/app/src/main/java/com/example/concordia_campus_guide/InfoCardFragment/InfoCardFragment.java#L91

In your case, you could have a setFloorplan(floorId) method that sets the floorplan according to a parameter passed to the function